psutil/_psutil_posix.c

```diff
@@ -429,6 +430,186 @@ psutil_net_if_mtu(PyObject *self, PyObject *args) {
     return PyErr_SetFromErrno(PyExc_OSError);
 }
 
+static bool
+check_and_append_iff_flag(PyObject *py_retlist, short int flags, short int flag_to_check, const char * flag_name)
+{
+    PyObject *py_str = NULL;
+
+    if (flags & flag_to_check) {
+        py_str = PyUnicode_DecodeFSDefault(flag_name);
+        if (! py_str)
+            return false;
+        if (PyList_Append(py_retlist, py_str)) {
+            Py_DECREF(py_str);
+            return false;
+        }
+        Py_CLEAR(py_str);
+    }
+
+    return true;
+}
+
+/*
+ * Get all of the NIC flags and return them.
+ */
+static PyObject *
+psutil_net_if_flags(PyObject *self, PyObject *args) {
+    char *nic_name;
+    int sock = -1;
+    int ret;
+    struct ifreq ifr;
+    PyObject *py_retlist = PyList_New(0);
+    short int flags;
+
+    if (py_retlist == NULL)
+        return NULL;
+
+    if (! PyArg_ParseTuple(args, "s", &nic_name))
+        goto error;
+
+    sock = socket(AF_INET, SOCK_DGRAM, 0);
+    if (sock == -1) {
+        PyErr_SetFromErrno(PyExc_OSError);
+        goto error;
+    }
+
+    PSUTIL_STRNCPY(ifr.ifr_name, nic_name, sizeof(ifr.ifr_name));
+    ret = ioctl(sock, SIOCGIFFLAGS, &ifr);
+    if (ret == -1) {
+        PyErr_SetFromErrno(PyExc_OSError);
```

Comment:
Please use `PyErr_SetFromOSErrnoWithSyscall("ioctl(SIOCGIFFLAGS)");`.

Reply:
Fixed in d83c959a981c68ff7ff3d60a155b4f35640e9fa9.